fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanTransactionRelationReadService.java

```diff
@@ -76,4 +81,41 @@ public List<LoanTransactionRelation> fetchLoanTransactionRelationFrom(final List
         return queryToExecute.getResultList();
     }
 
+    /**
+     * Determines if a transaction relation should be included in the API response.
+     *
+     * Filters out REPLAYED relations that are created during internal processing (e.g., accrual activities) but
+     * includes REPLAYED relations that are part of legitimate business operations (e.g., reverse-replay scenarios from
+     * backdated transactions).
+     *
+     * @param relation
+     *            the transaction relation to evaluate
+     * @return true if the relation should be included, false otherwise
+     */
+    private boolean shouldIncludeRelation(LoanTransactionRelation relation) {
+        LoanTransactionRelationTypeEnum relationType = relation.getRelationType();
+
+        // Always include non-REPLAYED relations
+        if (relationType != LoanTransactionRelationTypeEnum.REPLAYED) {
+            return true;
+        }
+
+        // For REPLAYED relations, include only those that are part of legitimate business operations
```

Comment:
why do we need this? i am not fully understand the reasoning behind. Having external id or not has nothing to do with reverse-replay scenarios...